activity.py

```diff
@@ -291,13 +284,12 @@ def _one_to_one_connection(self, tp_channel):
             return
         bus_name, connection, channel = json.loads(tp_channel)
         logger.debug('GOT XMPP: %s %s %s', bus_name, connection, channel)
-        Connection(bus_name, connection, ready_handler=lambda conn:
-                   self._one_to_one_connection_ready_cb(
-                       bus_name, channel, conn))
+        conn = TelepathyGLib.Connection.new(bus_name, connection)
+        self._one_to_one_connection_ready_cb(TelepathyGLib.DBusDaemon.dup(), bus_name, channel, conn)
 
     def _one_to_one_connection_ready_cb(self, bus_name, channel, conn):
         '''Callback for Connection for one to one connection'''
-        text_channel = Channel(bus_name, channel)
+        text_channel = TelepathyGLib.Channel(conn, channel)
```

Comment:
It's not astonishing to me.  It's very different code.  There was probably some mistake in `sugar3/presence/activity.py`.  Looking back at the context of the traceback, it was https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/401/commits/283dde0de137a1d5f4265bf1b700999c26025ef0 as-is, which was a mixed-binding cross-the-streams in-progress state of affairs.  Also, in making the next commit, I don't think @tonadev was considering why the argument count did not match, as the code was simply reverted.  I took it to be an experimental change to prove where the traceback happened.  I think I mentioned this before.

It has since changed in https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/412 as the line numbers don't match the traceback any more.

Is there any reason _not_ to use TelepathyGLib.Channel?  What happens if you try?

Reply:
> Is there any reason _not_ to use TelepathyGLib.Channel? What happens if you try?

When I try I `TelepathyGLib.Channel` in sugar-toolkit-gtk3 in the same way as used in chat, I get the above traceback, but when I use `TelepathyGLib.Channel.new` with proper arguments the traceback is gone.

`TelepathyGLib.Channel.new` returns a new Channel object, exactly what we want. What does 
``TelepathyGLib.Channel` return?

To initialize a new channel object isn't it necessary to use the `new` method?

Are these lines in the chat activity using `TelepathyGLib.Channel` gets hit?

Reply:
> > Is there any reason _not_ to use TelepathyGLib.Channel? What happens if you try?
> 
> When I try `TelepathyGLib.Channel` in sugar-toolkit-gtk3 in the same way as used in chat, I get the above traceback, but when I use `TelepathyGLib.Channel.new` with proper arguments the traceback is gone.

The traceback above is caused by exception _TypeError: GObject.__init__() takes exactly 0 arguments (2 given)_

Work on fixing the exception.  i.e. work on the argument list until the exception does not happen.

> `TelepathyGLib.Channel.new` returns a new Channel object, exactly what we want. What does
> `TelepathyGLib.Channel` return?

You can answer this yourself by testing.  Use `repr()` or `type()`.  Documentation also answers, just in case.

> To initialize a new channel object isn't it necessary to use the `new` method?

No, apparently not.  There are multiple methods.

```
>>> help(TelepathyGLib.Channel)
Help on class Channel in module gi.repository.TelepathyGLib:

class Channel(Proxy)
 |  :Constructors:
 |  
 |  ::
 |  
 |      Channel(**properties)
 |      new(conn:TelepathyGLib.Connection, object_path:str, optional_channel_type:str, optional_handle_type:TelepathyGLib.HandleType, optional_handle:int) -> TelepathyGLib.Channel
 |      new_from_properties(conn:TelepathyGLib.Connection, object_path:str, immutable_properties:dict) -> TelepathyGLib.Channel
 |  
\
```

> Are these lines in the chat activity using `TelepathyGLib.Channel` gets hit?

Do you mean are these lines executed?  That would be easy to test.  Try it?